aws/data_source_aws_route_test.go

```diff
@@ -73,69 +112,108 @@ func TestAccAWSRouteDataSource_LocalGatewayID(t *testing.T) {
 	})
 }
 
-func testAccDataSourceAwsRouteCheck(name string) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		rs, ok := s.RootModule().Resources[name]
-
-		if !ok {
-			return fmt.Errorf("root module has no resource called %s", name)
-		}
+func TestAccAWSRouteDataSource_CarrierGatewayID(t *testing.T) {
+	dataSourceName := "data.aws_route.test"
+	resourceName := "aws_route.test"
+	rName := acctest.RandomWithPrefix("tf-acc-test")
 
-		r, ok := s.RootModule().Resources["aws_route.test"]
-		if !ok {
-			return fmt.Errorf("can't find aws_route.test in state")
-		}
-		rts, ok := s.RootModule().Resources["aws_route_table.test"]
-		if !ok {
-			return fmt.Errorf("can't find aws_route_table.test in state")
-		}
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSWavelengthZoneAvailable(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSRouteDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSRouteDataSourceConfigIpv4CarrierGateway(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(resourceName, "destination_cidr_block", dataSourceName, "destination_cidr_block"),
+					resource.TestCheckResourceAttrPair(resourceName, "route_table_id", dataSourceName, "route_table_id"),
+					resource.TestCheckResourceAttrPair(resourceName, "carrier_gateway_id", dataSourceName, "carrier_gateway_id"),
+				),
+			},
+		},
+	})
+}
 
-		attr := rs.Primary.Attributes
+func TestAccAWSRouteDataSource_DestinationPrefixListId(t *testing.T) {
+	dataSourceName := "data.aws_route.test"
+	resourceName := "aws_route.test"
+	rName := acctest.RandomWithPrefix("tf-acc-test")
 
-		if attr["route_table_id"] != r.Primary.Attributes["route_table_id"] {
-			return fmt.Errorf(
-				"route_table_id is %s; want %s",
-				attr["route_table_id"],
-				r.Primary.Attributes["route_table_id"],
-			)
-		}
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSRouteDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSRouteDataSourceConfigPrefixListNatGateway(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(resourceName, "destination_prefix_list_id", dataSourceName, "destination_prefix_list_id"),
+					resource.TestCheckResourceAttrPair(resourceName, "nat_gateway_id", dataSourceName, "nat_gateway_id"),
+					resource.TestCheckResourceAttrPair(resourceName, "route_table_id", dataSourceName, "route_table_id"),
+				),
+			},
+		},
+	})
+}
 
-		if attr["route_table_id"] != rts.Primary.Attributes["id"] {
-			return fmt.Errorf(
-				"route_table_id is %s; want %s",
-				attr["route_table_id"],
-				rts.Primary.Attributes["id"],
-			)
-		}
+func TestAccAWSRouteDataSource_GatewayVpcEndpoint(t *testing.T) {
```

Comment:
This test validates [this change](https://github.com/hashicorp/terraform-provider-aws/pull/17295/files#diff-b8af03892138451662b5730171b62cc571d6964da7026db591b276a37c5dc81fR123-R127) by adding a Gateway VPC Endpoint (S3), specifying that endpoint's prefix list ID to the data source and verifying that no route is returned.